mesonbuild/interpreter/interpreter.py

```diff
@@ -2397,7 +2397,7 @@ def func_install_subdir(self, node: mparser.BaseNode, args: T.Tuple[str],
                   validator=lambda x: 'must be `false` if boolean' if x is True else None),
         OUTPUT_KW,
         KwargInfo('output_format', str, default='c', since='0.47.0',
-                  validator=in_set_validator({'c', 'nasm'})),
+                  validator=in_set_validator({'c', 'nasm', 'custom'})),
```

Comment:
This is added without documentation and without a FeatureNew (see `since_values` in the KwargInfo docstring), which regardless of anything else is a hard blocker here.

As far as the new feature goes, this basically feels like ~~abusing~~ reusing configure_file for what might be better named as `import('fs').write(fname, data)`, which thematically matches the existing `fs.read()`.

And it also follows the trend of `configure_file(..., copy: true)` of being... honestly? a really bad API. See #10042 for an attempt at deprecating that as a bad idea and replacing it with a proposed new `fs.copy()`. I don't like the idea of adding more functionality to configure_file that overloads the existing semantics in order to do too many disparate things.

Reply:
The issue as I see it is that this MR has two separate features in it. The first is adding custom multiline text to generated files and the second one is this custom file type. The former we definitely want but the latter is a bit confusing. Why do you actually need it? The use case seems to be for generating headers and for that you can already use `configure_file` and `append_raw_text` without an input file. The only downside is that you get the `#pragma once` bit but that is something you really want to have in headers.

Reply:
@eli-schwartz @jpakkane I'll get rid of this additional `custom` format in this MR to go with the base feature.

The use case for `custom` was basically the ability to easily generate files from scratch. I agree it can be considered as wrong use of the API, as I realized that I could find no other easy way to do it right.
I'll look anyway at the `fs` module (though I guess we must add a bit more meson code to handle file installation). It might just be what I needed then.

In any case, just give me a few days to get back to this MR and bring it back to the original fix.

Reply:
The new "custom" output_format was removed from this MR. Let's get back to the root issue. I may pursue the new proposed feature in a dedicated MR later.